cishouseholds/impute.py

```diff
@@ -348,18 +348,17 @@ def impute_and_flag(df: DataFrame, imputation_function: Callable, reference_colu
     df = imputation_function(
         df, column_name_to_assign="temporary_imputation_values", reference_column=reference_column, **kwargs
     )
-
     status_column = reference_column + "_is_imputed"
-    status_other = F.col(status_column) if status_column in df.columns else None
+    status_other = F.col(status_column) if status_column in df.columns else F.lit(None)
 
     df = df.withColumn(
         status_column,
-        F.when(F.col("temporary_imputation_values").isNotNull(), 1)
+        F.when(status_other == 1, 1)
```

Comment:
comparison sometimes seems to require a `Column` object so I think we'd need to 
```suggestion
        F.when(status_other == F.lit(1), 1)
```

Reply:
Scratch this, I think you've fixed this above